README.md

```diff
@@ -4,12 +4,57 @@ XenAPI Software Development Kit
 This repository contains the generation code for the XenAPI Software Development
 Kit (SDK) and its associated documentation and examples.
 
-The generated SDK consists of five components, one for each of C, C#, Java,
-PowerShell and Python, exposing the individual XenAPI calls as first-class
-functions in the target language.
+The SDK consists of five components, one for each of C, C#, Java, PowerShell,
+and Python, exposing the individual XenAPI calls as first-class functions in the
+target language.
 
-The generation code is mostly written in
-[Ocaml](http://caml.inria.fr/ocaml/) 3.12.
+The generation code is mostly written in [Ocaml][1] 3.12.
+
+Build
+-----
+
+#### 1. Environment setup
+
+Install [opam][2]. It is also recommended to install one of the listed external
+solvers (e.g. aspcud).
+
+Ensure you are using version 4.02.3 of the Ocaml compiler:
+`opam switch 4.02.3`
+
+and you have the package depext: `opam install depext`
+
+_Note: if you want to avoid setting up the environment by yourself, you can use
+one of the ready containers from [opam-dockerfiles][3], which come with a
+preinstalled OPAM environment and Ocaml compiler._
+
+#### 2. Install dependencies
+
+Add [xs-opam][4] as a remote Opam repository (follow the instructions in that
+repo).
+
+The SDK is dependent on the XenAPI, therefore run
+
+```
+   opam depext -y xapi
+   opam install -y xapi
```

Comment:
I would think this belongs to the xapi repo since  it's a xapi build problem; we can refer the users there for any problems they have.